src/Integrations/IntegrationRegistry.php

```diff
@@ -0,0 +1,179 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\Integrations;
+
+/**
+ * Class used for tracking registered integrations with various Block types.
+ */
+class IntegrationRegistry {
+	/**
+	 * Integration identifier is used to construct hook names and is given when the integration registry is initialized.
+	 *
+	 * @var string
+	 */
+	protected $registry_identifier = '';
+
+	/**
+	 * Registered integrations, as `$name => $instance` pairs.
+	 *
+	 * @var IntegrationInterface[]
+	 */
+	protected $registered_integrations = [];
+
+	/**
+	 * Initializes all registered integrations.
+	 *
+	 * Integration identifier is used to construct hook names and is given when the integration registry is initialized.
+	 *
+	 * @param string $registry_identifier Identifier for this registry.
+	 */
+	public function initialize( $registry_identifier = '' ) {
+		if ( $registry_identifier ) {
+			$this->registry_identifier = $registry_identifier;
+		}
+
+		if ( empty( $this->registry_identifier ) ) {
```

Comment:
This should be first in the function block.

Reply:
The conditional above this sets `$this->registry_identifier` based on the incoming class constructor param, so I think this ism in the correct place.

Reply:
I should have been more explicit. I am talking about the **return early** pattern.

There is no reason to set $this->registry_identifier if $registry_identifier is an empty sting.

```php
if ( empty( $registry_identifier ) ) {
    _doing_it_wrong( __METHOD__, esc_html( __( 'Integration registry requires an identifier.', 'woo-gutenberg-products-block' ) ) );
		return false;
}

$this->registry_identifier = $registry_identifier;
```

Reply:
I see. The thing is, this works so that `$this->registry_identifier` may be set on the class itself. So it needs to check both $this->registry_identifier and $registry_identifier once resolve. Does that make sense?

Reply:
Ah OK, now I understand. If `$registry_identifier` is set it overwrites what is in `$this->reg...`. Later we just check if we have anything either set on the class or set through the function parameter. Thank you for the explanation.